libs/domains/organizations/feature/src/lib/git-repository-setting/git-repository-setting.tsx

```diff
@@ -0,0 +1,88 @@
+import { type GitProviderEnum, type GitRepository } from 'qovery-typescript-axios'
+import { Controller, useFormContext } from 'react-hook-form'
+import { useParams } from 'react-router-dom'
+import { InputSelect, LoaderSpinner } from '@qovery/shared/ui'
+import { upperCaseFirstLetter } from '@qovery/shared/util-js'
+import { getGitTokenValue } from '../git-provider-setting/git-provider-setting'
+import useRepositories from '../hooks/use-repositories/use-repositories'
+
+export interface GitRepositorySettingProps {
+  gitProvider: GitProviderEnum | string
+  disabled?: boolean
+}
+
+export function GitRepositorySetting({ disabled, gitProvider }: GitRepositorySettingProps) {
+  const { control, setValue, watch } = useFormContext()
+  const { organizationId = '' } = useParams()
+
+  const getGitToken = getGitTokenValue(gitProvider)
+  const gitProviderOrTokenType = getGitToken ? getGitToken?.type : gitProvider
+  const watchFieldRepository = watch('repository')
+
+  const {
+    data: repositories = [],
+    isLoading,
+    isRefetching,
+    isError,
+  } = useRepositories({
+    organizationId,
+    gitProvider: gitProviderOrTokenType,
+    gitToken: getGitToken?.id,
+    enabled: !disabled,
+  })
+
+  if (isError) {
+    return null
+  }
+
+  if (!disabled && (isLoading || isRefetching)) {
+    return (
+      <div data-testid="loader" className="flex justify-center">
```

Comment:
Is this data-testid still needed?

Reply:
Not needed, removed https://github.com/Qovery/console/pull/913/commits/5f0ff27fbf70a12706bd3d915ea9360cbec3cc55